dev/com.ibm.ws.microprofile.faulttolerance.2.0/test/src/com/ibm/ws/microprofile/faulttolerance/test/AsyncTest.java

```diff
@@ -204,6 +202,46 @@ public void testAsyncRequestContextDeactivatedOnInterruption() throws Interrupte
         }
     }
 
+    @Test
+    public void testAsyncRequestContextDeactivatedOnInterruption() throws InterruptedException, ExecutionException, TimeoutException, TestException {
+        ExecutorBuilder<String> builder = FaultToleranceProvider.newExecutionBuilder();
+
+        MockAsyncRequestContextController asyncRequestContextController = new MockAsyncRequestContextController();
+        builder.setRequestContextController(asyncRequestContextController);
+
+        Executor<Future<String>> executor = builder.buildAsync(Future.class);
+        ExecutionContext context = executor.newExecutionContext("testAsyncCS", null);
+
+        CountDownLatch startSignal = new CountDownLatch(1); // Prevents the execution from being interrupted until request context is set
+
+        // Test an execution which is cancelled
+        Future<String> result = executor.execute(() -> {
+            assertEquals("Activate request context should be called when request is started", 1, asyncRequestContextController.getActivateContextCount());
```

Comment:
This assert won't cause the test to fail correctly as it's inside the executor's callable and the result of that is thrown away because you cancel the execution.

However, if this assert ever did fail, you wouldn't call `countDown()` on line 222 and then your test would hang waiting for it on line 231.

I don't really think this check is necessary as you're already testing this elsewhere, so you can just take it out.

Reply:
Okay that's great, thanks for the explanation